Inspicio/Controllers/ImagesController.cs

```diff
@@ -270,59 +274,105 @@ public class RatingBody
+                        if (image.NoOfRejections > 0)
                         {
-                            image.NoOfDislikes--;
+                            image.NoOfRejections--;
                         }
                     }
-                    // if both buttons are false, increment likes by 1
-                    else if (review.Liked == false)
+                    else if ((review.Approved == false) && (review.NeedsWork == false) && (review.Rejected == false))
                     {
-                        review.Liked = true;
-                        review.Disliked = false;
-                        image.NoOfLikes++;
+                        review.Approved = true;
+                        image.NoOfApprovals++;
                     }
                 }
             }
 
-            // if the dislike button has been pressed
-            else
+            // if the review has been rejected
+            if (data.ThumbChosen == (int)Status.REJECTED)
             {
-                // if dislike button hasn't been pressed before
-                if (review.Disliked == false)
+                if (review.Rejected == false)
                 {
-                    // if the like button was pressed add 1 to dislikes, minus 1 from likes
-                    if (review.Liked == true)
+                    if (review.NeedsWork == true)
                     {
-                        review.Liked = false;
-                        review.Disliked = true;
-                        image.NoOfDislikes++;
-                        if (image.NoOfLikes > 0)
+                        review.Rejected = true;
+                        review.NeedsWork = false;
+                        image.NoOfRejections++;
+                        if (image.NoOfNeedsWork > 0)
                         {
-                            image.NoOfLikes--;
+                            image.NoOfNeedsWork--;
                         }
                     }
-                    // if neither button has been pressed, add 1 to dislikes
-                    else if (review.Liked == false)
+                    else if (review.Approved == true)
                     {
-                        review.Disliked = true;
-                        image.NoOfDislikes++;
+                        review.Approved = false;
+                        review.Rejected = true;
+                        image.NoOfRejections++;
+                        if (image.NoOfApprovals > 0)
+                        {
+                            image.NoOfApprovals--;
+                        }
+                    }
+                    else if ((review.Approved == false) && (review.NeedsWork == false) && (review.Rejected == false))
+                    {
+                        review.Rejected = true;
+                        image.NoOfRejections++;
                     }
                 }
             }
 
+            // if the review needs more work
+            if (data.ThumbChosen == (int)Status.NEEDSWORK)
```

Comment:
I was expecting the `ThumbChosen` property to no longer be an int now that the enum exists.